### PR TITLE
Sites: Estimate terrain objects bounding box overlap with site for hiding rocks etc

### PR DESCRIPTION
**Pull request by dijksterhuis**
TL;DR stop mortars spawned inside rocks soft locking a zone by removing ROCK/ROCKS/HIDE terrain objects with a bounding box estimation.

Estimates whether nearby terrain objects will overlap/intersect with the spawned site, then hides them. Also hides any terrain objects directly on top of those found with bounding box search, otherwise we end up with a bunch of floating bushes. Can be expanded to other kinds of terrain objects on a per site basis.

Requires more testing on my end for both functionality and performance ... so putting up as a draft PR for now.

### TODO

- [ ] Use `getDir` on terrain objects to more accurately calculate overlap/intersections (rectangular area instead of circular?)
- [ ] Test a different `clippingType` argument for `boundingBoxReal` https://community.bistudio.com/wiki/boundingBoxReal
- [ ] Calculate site radius using the object with largest 2D distance from site centre point.

## screenshots

notes about screenshots

- used different parameters to what is in this PR. `_siteRadius` for this example would have been 15, not 5.
- the large rock terrain object that was hidden has a large maximum dimension, which is why it is hidden even though it's quite far away from the site. 
- rocks on top of other rocks can be problematic, the secondary search with `nearestTerrainObjects` might want yet another bounding box calculation

### debug markers

image

### terrain hiding

image

### CODE REVIEW

**Comment by dijksterhuis** [comment]
Please can someone confirm that `vn_mf_sites_load` is non-functional? 

Did a `grep` for wherever the sites data is supposed to get set in the profile DB but came up with nothing.

```bash
$ find . -name "*.sqf" -exec grep -Hn 'call para_s_fnc_profile_db' {} \; | grep sites
./mission/functions/systems/sites/fn_sites_load.sqf:19:(["GET", "sites", []] call para_s_fnc_profile_db) params ["","_sitesData"];
```

File: mission/functions/systems/sites/fn_sites_hide_terrain_objects_bbox_intersections.sqf [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+			_marker setMarkerSize [_maxDimension, _maxDimension];
+			_marker setMarkerBrush "Border";
+
+			_marker setMarkerColor "ColorRed";
+
+		};
+	};
+};
+
+_objsIntersectTerrainObjectsBBox
```
Comment (dijksterhuis) [inline mission/functions/systems/sites/fn_sites_hide_terrain_objects_bbox_intersections.sqf:171]: :eyes: this is definitely wrong and needs fixing.

File: mission/functions/systems/sites/fn_sites_generate.sqf [hunk 2]
```diff
@@ -54,13 +54,18 @@ private _fnc_findPos = {
 	//Create initial AA emplacements
 	for "_i" from 1 to (1 + ceil random (vn_mf_s_max_aa_per_zone - 1)) do
 	{
-		[[_center, _sizeMax / 4, _sizeMax / 2] call _fnc_findPos] call vn_mf_fnc_sites_create_aa_site;
+		private _pos = [_center, _sizeMax / 4, _sizeMax / 2] call _fnc_findPos;
+
+		[_pos] call vn_mf_fnc_sites_create_aa_site;
+		[_pos, 5] call vn_mf_fnc_sites_hide_terrain_objects_bbox_intersections;
```
Comment (veteran29) [inline mission/functions/systems/sites/fn_sites_generate.sqf:60]: Is this part of MF code scheduled? If it is you should hide terrain objects before spawning.
